almost_tcp/message_hdl.py

```diff
@@ -0,0 +1,249 @@
+"""
+Almost TCP data structures: messages and packet decoder.
+
+Almost TCP is a protocol designed to tunnel data streams
+(specifically, HTTP requests) over a serial port.
+It consists of a subset of TCP's functionality,
+including reduced sizes for some fields,
+with the goal of enabling a small hardware implementation.
+
+Each ATCP message consists of a fixed-length header
+followed by variable-length data.
+
+All numeric fields are network-endian (big-endian).
+
+-   Connection-oriented: syn/synack/ack setup
+-   Single stream number, not two port numbers
+    -   Stream number probing to determine # of hardware resources available
+
+"""
+
+from amaranth import Module, Signal, unsigned, Const
+from amaranth.lib.wiring import Component, In, Out, Signature
+from amaranth.lib import stream
+from amaranth.lib.data import UnionLayout, ArrayLayout, Struct
+from amaranth.lib.fifo import SyncFIFOBuffered
+
+
+class FlagsLayout(Struct):
+    """
+    Layout of the flags in an ATCP header.
+    """
+
+    # Indicates the sender has sent all the data it will send on this stream.
+    fin: 1
+    # Indicates a desire to open a new stream,
+    # starting at 1 more than the provided sequence number.
+    syn: 1
+    # Indicates the provided stream should be immediately reset.
+    rst: 1
+    # Unused: push data to the receiver.
+    psh: 1
+    # The acknowledgement number is significant.
+    ack: 1
+    # Unused: urgent
+    urg: 1
+    # Unused: explicit congestion notification
+    ecn: 1
+    # Unused: congestion window reduced
+    cwr: 1
+
+
+class HeaderLayout(Struct):
+    """
+    Layout of a full ATCP header.
+
+    TODO: Fix byte order!
```

Comment:
What's the issue with the byte order?

Reply:
Uh...nothing, any more, I think?

Yeah, I had been handling it wrong (using little-endian on the chip), but fixed it in the implementation without updating this comment. Removed.